ecommerce/core/management/commands/tests/test_download_sdn_fallback.py

```
@@ -0,0 +1,43 @@
+"""
+Tests for Django management command to download csv for SDN fallback.
+"""
+from django.core.management import call_command
+from django.core.management.base import CommandError
+from mock import patch
+
+from ecommerce.tests.testcases import TestCase
+
+
+class TestDownloadSndFallbackCommand(TestCase):
+
+    def setUp(self):
```

Should there be a test case for when it 500s (e.g. the server hosting the CSV is down/the file got renamed)?

---

I played around with that a little: since we're just hitting a url we get back a response (rather than a csv itself), so if we hit a junk url or get an error, we get much smaller response that doesn't meet the threshold if we save into a temp csv. Do we want to add tests that mock different kinds of bad responses?

---

I would imagine we would want to know a) how we handle HTTP issues with downloads and 2) if someone changed the code and either started hiding the 500s or made it fail even less gracefully

---

@emma-green @inventhouse This download ticket just asked to error if the download is below the threshold- do we want to add requirements about bubbling up some clues about why it errored, such as searching the response for a status code, etc?

---

I'm not sure that we need to specifically *handle* those cases in the code itself(although I could be convinced), but I definitely see the utility in adding a test case for what we *expect* to happen in those cases (throw an error).

---

If I correctly interpret Emma's comment above, I would agree. I don't think we need to have to write a test case or catch+do something specific for every specific HTTP status code that comes back, but it would be good to have a test case to handle a generic "what happens if there is an unexpected (aka non-2xx http status) response from the government website" so we know what to expect (e.g. a warning log statement? everything breaks? error is ignored?).

---

thank you both! Today if we do get back a response with a non-2xx status, we fall under the exception that our download didn't meet the threshold, and I've added logger.warning with the actual response that came back (shows a 503 when it's a 503, etc), so we'll be able to see it. Then I've also added a test to cover the case where we don't even get THAT much back (realized that was missing)- if we hit a nonsense url we get a ConnectionError, now that's a different CommandError. Let me know if anything else would be helpful